android/build.gradle

```diff
@@ -43,7 +43,31 @@ if (isNewArchitectureEnabled()) {
 }
 apply plugin: 'kotlin-android'
 
+def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')[0].toInteger()
+def shouldUseNameSpace = agpVersion >= 7
+def PACKAGE_PROP = "package=\"com.reactnativecommunity.webview\""
+def manifestOutFile = file("${projectDir}/src/main/AndroidManifest.xml")
+def manifestContent = manifestOutFile.getText()
+if(shouldUseNameSpace){
+      manifestContent = manifestContent.replaceAll(
+        PACKAGE_PROP,
+        ''
+    )  
+} else {
+    if(!manifestContent.contains("$PACKAGE_PROP")){
+        manifestContent = manifestContent.replace(
+            '<manifest',
+            "<manifest $PACKAGE_PROP "
+        )
+    }
+}
+manifestContent.replaceAll("  ", " ")
+manifestOutFile.write(manifestContent)
```

Comment:
This is a lot more than the "two one-liners" mentioned in the description. Could you add a comment explaining the purpose of this? (I expect at some point we'll be able to remove it as we drop support for older React Native versions)

Reply:
This might be a leaner fix:
https://github.com/react-native-community/discussions-and-proposals/issues/671#issuecomment-1598892444

Reply:
Indeed. I did read the thread though and understand that the point of the more complicated version is to reduce compiler warnings... which isn't a bad idea, if it can be done in an understandable way.

Reply:
@TheAlmightyBob if user can accept the warning message, we can keep package in AndroidManifest and add namspace in build.gradle. Then we can keep going